airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/unstructured_parser.py

```diff
@@ -118,6 +118,8 @@ def _read_file(self, file_handle: IOBase, remote_file: RemoteFile, format: Unstr
             file = BytesIO(file_handle.read())
             file_handle.seek(0)
             elements = unstructured_partition_pdf(file=file)
+            # need to close the inner file object to prevent memory leak
+            file.close()
```

Comment:
This pops out as potentially being a good use case for using a context manager, which is slightly more readable and will close the file even if there's an exception.

```suggestion
            with BytesIO(file_handle.read()) as file:
                file_handle.seek(0)
                elements = unstructured_partition_pdf(file=file)
```

That said, with a (very short) Google search, I didn't find definitive documentation to confirm that `BytesIO` supports this pattern. Feel free to disregard if not helpful - the slight change in design pattern isn't worth extra scope, if it requires more refactoring or if we're not sure if it gives the right result.

Reply:
That's a good suggestion, thanks! I will adjust